java-checks/src/test/files/checks/FinalClassCheck.java

```diff
@@ -49,3 +49,23 @@ public static int magic(){
 
 public class NoConstructorClass { // Compliant, implicit constructor has package visibility
 }
+
+public abstract class AbstractClass {  // Compliant, constructor is used in
```

Comment:
This test lacks the case of an anonymous class : new AbstractClass("") {}  

I am  not sure if your code will work properly with this case or not so it deserves to be tested at least.